program fgetu_test;
#include( "stdlib.hhf" )

var
	Handle		:dword;
	inputF		:real80;
	
begin fgetu_test;


	// Testing fileio.getu32
	
	try
	
		fileio.open( "fgetu32.data", fileio.r );
		mov( eax, Handle );

		fileio.getu32( Handle );
		if( eax <> 1 ) then
		
			raise(1011);
		
		endif;	
		
		fileio.getu32( Handle );
		if( eax <> 4294967295 ) then
		
			raise(1012);
		
		endif;	
				
		fileio.getu32( Handle );
		if( eax <> 2147483647 ) then
		
			raise(1014);
		
		endif;


		
		try
		
			fileio.getu32( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		fileio.readLn( Handle );	// Clean up after ourselves.
		
		
		try
		
			fileio.getu32( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1015 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1016 );
			
		endtry;
		fileio.readLn( Handle );
		
		
		// Better be at end of file.
		
		if( !fileio.eof( Handle )) then
		
			raise(1010);
			
		endif;
		stderr.put( "fgetu32 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"fgetu32 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;








	// Testing fileio.getu16
	
	try
	
		fileio.open( "fgetu16.data", fileio.r );
		mov( eax, Handle );

		fileio.getu16( Handle );
		if( eax <> 0 ) then
		
			raise(1021);
		
		endif;	
		
		fileio.getu16( Handle );
		if( eax <> 1 ) then
		
			raise(1022);
		
		endif;	
		
		fileio.getu16( Handle );
		if( eax <> 65535 ) then
		
			raise(1024);
		
		endif;	
		
		fileio.getu16( Handle );
		if( eax <> 32767 ) then
		
			raise(1025);
		
		endif;
		
		try
		
			fileio.getu16( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		fileio.readLn( Handle );	// Clean up after ourselves.
		
		
		try
		
			fileio.getu16( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1026 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1027 );
			
		endtry;
		fileio.readLn( Handle );
		
		
		// Better be at end of file.
		
		if( !fileio.eof( Handle )) then
		
			raise(1020);
			
		endif;
		stderr.put( "fgetu16 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"fgetu16 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;





	// Testing fileio.getu8
	
	try
	
		fileio.open( "fgetu8.data", fileio.r );
		mov( eax, Handle );

		fileio.getu8( Handle );
		if( eax <> 0 ) then
		
			raise(1031);
		
		endif;	
		
		fileio.getu8( Handle );
		if( eax <> 1 ) then
		
			raise(1032);
		
		endif;	
		
		fileio.getu8( Handle );
		if( eax <> 255 ) then
		
			raise(1033);
		
		endif;	
		
		fileio.getu8( Handle );
		if( eax <> 127 ) then
		
			raise(1035);
		
		endif;
		
		try
		
			fileio.getu8( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		fileio.readLn( Handle );	// Clean up after ourselves.
		
		
		try
		
			fileio.getu8( Handle );

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1036 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1037 );
			
		endtry;
		fileio.readLn( Handle );
		
		
		// Better be at end of file.
		
		if( !fileio.eof( Handle )) then
		
			raise(1030);
			
		endif;
		stderr.put( "fgetu8 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"fgetu8 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;




			
end fgetu_test;

													